presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/MergeLimitWithSort.java

```diff
@@ -51,6 +51,6 @@
                         parent.getCount(),
                         child.getOrderBy(),
                         child.getOrderings(),
-                        parent.isPartial()));
+                        parent.isPartial() ? TopNNode.Step.PARTIAL : TopNNode.Step.SINGLE));
```

Comment:
I did something similar for Limit node here: https://github.com/prestodb/presto/pull/6700/commits/07f7fb84f2da3a798ff011abaf6b1247a104bba9 (still pending)